src/stores/AlertStore.ts

```diff
@@ -0,0 +1,53 @@
+import { action, computed, makeObservable, observable } from 'mobx'
+import { enableStaticRendering } from 'mobx-react'
+
+enableStaticRendering(typeof window === 'undefined')
+
+interface Alert {
+  id: number
+  show: boolean
+  type: AlertType
+  message: string
+  duration?: number
+}
+type AlertType = 'error' | 'warning' | 'info' | 'success'
+
+export class AlertStoreImpl {
+  alerts: Alert[] = []
+
+  constructor() {
+    makeObservable(this, {
+      alerts: observable,
+      show: action,
+      hide: action,
+      clear: action,
+      getAlerts: computed,
+    })
+  }
+
+  show(message: string, type: AlertType, duration = 3000) {
+    this.clear()
+    const alert: Alert = {
+      id: +new Date(),
+      show: true,
+      message,
+      type,
+      duration,
+    }
+    this.alerts.push(alert)
+  }
+
+  hide() {
+    if (this.alerts.length > 0) this.alerts[0].show = false
```

Comment:
We might pass an `id` to the hide method, so we hide a specific alert, instead of the first one in the list.

```suggestion
  hide(id: number) {
    if (this.alerts.length > 0) this.alerts[id].show = false
```